File: app/src/main/java/com/og/emdlc/dagger/AbstractTest.java
package com.og.emdlc.dagger;

import android.content.Context;

import com.og.emdlc.application.EmdlcApplication;
import com.og.emdlc.managers.AccountManager;

import javax.inject.Inject;

/**
 * Created by olivier.goutay on 1/25/17.
 */
public class AbstractTest {

    @Inject
    protected AccountManager mAccountManager;

    /**
     * Inject all the Managers/Dependencies you want to use in tests here
     */
    protected void beforeActivityLaunched(Context context) {
        ((EmdlcApplication) context.getApplicationContext()).getDaggerComponent().inject(this);
    }

    /**
     * Used to reset the injection
     */
    protected void tearDown(Context context) {
        ((EmdlcApplication) context.getApplicationContext()).setDaggerComponent(null);
    }
}

File: app/src/main/java/com/og/emdlc/dagger/AbstractTest.kt
package com.og.emdlc.dagger

import android.content.Context

import com.og.emdlc.application.EmdlcApplication
import com.og.emdlc.managers.AccountManager

import javax.inject.Inject

/**
 * Created by olivier.goutay on 1/25/17.
 */
open class AbstractTest {

    @Inject
    lateinit var accountManager: AccountManager

    /**
     * Inject all the Managers/Dependencies you want to use in tests here
     */
    protected fun beforeActivityLaunched(context: Context) {
        EmdlcApplication.daggerComponent.inject(this)
    }

}
